docs/enhancements/api-for-skipping-validations.md

```diff
@@ -0,0 +1,89 @@
+---
+title: api-for-skipping-validations
+authors:
+  - "@pmaidmen"
+creation-date: 2023-01-08
+last-updated: 2023-01-08
+---
+
+# Allow validations to be skipped
+
+# Summary
+
+We need a new api that will allow us to skip cluster/host validations, this will be for unsupported use.
+This API should work on a per cluster basis and will have it's own feature flag to allow us to enable it for specific users.
+An event should be emitted whenever the API is used during an installation.
+Clusters containing skipped validations should not contribute to statistics as these may be skewed.
+# Motivation
+
+Some users have very specific HW that doesn't pass our validations and we want to allow them to proceed with installation.
+Sometimes we have bugs in our validations that block people from installing and we don't want our partners left in a position where they can't install.
+
+### Goals
+
+- Allow validations to be bypassed so that installation may proceed.
+
+### Non-Goals
+
+- None identified at this stage
+
+### The implementation of skipped-validations
+
+The way we propose to deal with the validation is as follows
+
+#### Propoal - Run all validations, allow the user to review but proceed anyway. (Suggested by Michael Hrivnak as an intepretation of requirements)
+
+    * All validations run.
+    * One or more validations fail and describe a problem.
+    * The user evaluates the alleged problems and decides to try the installation anyway.
+    * When skipping validations, there will be some "absolutely mandatory" validations that cannot fail prior to proceeding.
+        - This list is TBC
+    * If the user skips validations (by making a change to a boolean in the cluster object), we should
+        * Emit an event
+        * Record the validation skip in the cluster in some way
+        * Log the validations that were failing at the moment the validations were skipped.
+          - Maybe add this as a field to the status of the cluster.
+
+### Effect on Triage tickets.
+To ensure that there is less noise for Triage tickets, any cluster that has events consistent with this sort of override should be omitted from Triage ticketing, in a similar way to how this is done for jupiter installations.
+
+Example here: https://github.com/openshift-assisted/assisted-installer-deployment/blob/463dc6612c923c46d47e80762c72651758a690d3/tools/create_triage_tickets.py#L64
+
+### Per customer feature enablement.
+This feature is to be enabled on a per customer basis, so will need to use the "per customer feature enablement" feature.
+
+See the following link for more information.
+
+https://github.com/openshift/assisted-service/blob/master/docs/dev/feature-per-customer.md
+
+### User Stories
+
+#### Story 1
+
+As a user with known validation issues that need to be bypassed, I would like to be able to bypass VIP validation in cases where VIPS could be incorrectly shown as busy. So that I may proceed with installation.
```

Comment:
```suggestion
As a user with known validation issues that need to be bypassed, I would like to be able to bypass VIP validation in cases where VIPs are incorrectly shown as busy so that I may proceed with the installation.
```